cypress/e2e/specialFeatures/atiAnalytics/assertions/index.js

```diff
@@ -103,78 +133,188 @@ export const assertPageView = ({
   });
 };
 
-const getViewClickDetailsRegex = ({ contentType, component, pageIdentifier }) =>
-  new RegExp(
-    `PUB-\\[${contentType}(.*)?\\]-\\[${component}(.*)?\\]-\\[(.*)?\\]-\\[(.*)?\\]-\\[${pageIdentifier}\\]-\\[(.*)?\\]-\\[(.*)?\\]-\\[(.*)?\\]`,
-    'g',
+const assertCPVModelViewEvent = ({
+  component,
+  pageIdentifier,
+  contentType,
+  useReverb,
+  params,
+}) => {
+  assertATIComponentViewEventParamsExist({ params, useReverb });
+
+  if (!useReverb) {
+    expect(params.p).to.equal(pageIdentifier, 'params.p (page identifier)');
+  }
+
+  expect(params.ati).to.match(
+    getViewClickDetailsRegex({
+      contentType,
+      component,
+      pageIdentifier,
+    }),
+    'params.ati (publisher impression)',
   );
+};
+
+const assertViewabilityModelViewEvent = ({
+  component,
+  pageIdentifier,
+  contentType,
+  params,
+}) => {
+  const eventContext = JSON.parse(params.context);
+
+  assertReverbViewabilityComponentEventParamsExist({
+    params,
+  });
+
+  expect(params.events).to.match(
+    getViewabilityEventDetailsRegex({
+      contentType,
+      component,
+      actionType: 'view',
+    }),
+    'params.events (publisher impression)',
+  );
+
+  expect(eventContext[0].data.page.$).to.equal(pageIdentifier);
+};
 
 export const assertATIComponentViewEvent = ({
   component,
   pageIdentifier,
   contentType,
   useReverb,
-}) =>
-  cy
-    .wait(`@${component}-ati-view`)
+}) => {
+  const useViewabilty = usesReverbViewabilityModel({
+    useReverb,
+    appEnv: Cypress.env('APP_ENV'),
```

Comment:
What don't we just remove appEnv as a param, and check within the `usesReverbViewabilityModel` function instead?